prov/slurm/execution.go

```diff
@@ -426,25 +428,34 @@ func (e *executionCommon) prepareAndSubmitJob(ctx context.Context) error {
 	var cmd string
 	if e.jobInfo.Command != "" {
 		inner := fmt.Sprintf("%s %s", e.jobInfo.Command, quoteArgs(e.jobInfo.Args))
-		cmd = e.wrapCommand(inner)
+		var err error
+		cmd, err = e.wrapCommand(inner)
+		if err != nil {
+			return err
+		}
 	} else {
 		cmd = fmt.Sprintf("%s%ssbatch -D %s%s %s", e.addWorkingDirCmd(), e.buildEnvVars(), e.jobInfo.WorkingDir, e.buildJobOpts(), path.Join(e.jobInfo.WorkingDir, e.PrimaryFile))
 	}
 	return e.submitJob(ctx, cmd)
 }
 
-func (e *executionCommon) wrapCommand(innerCmd string) string {
-	pathScript := path.Join(e.jobInfo.WorkingDir, batchScript)
+func (e *executionCommon) wrapCommand(innerCmd string) (string, error) {
+	id, err := uuid.NewRandom()
```

Comment:
Couldn't we use e.deploymentID  + e.nodeName + e.taskID + e.operation.Name ?
(not sure if there is this lack of entropy issue with uuid.newRandom() that we are seeing sometimes on Ansible)

Reply:
I think it will be more complicated to use  `e.deploymentID + e.nodeName + e.taskID + e.operation.Name`

1. This will be very very long as taskID is it self a UUID
2. We should deal with not file name friendly characters in node name and operation name

We already generate UUIDs so I think we have enough entropy.